infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -128,6 +130,9 @@ public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
             throw new LogUnitException(msg);
         }
 
+        String codec = (String) serverContext.getServerConfig().get("--compression-codec");
+        compressionCodec = Codec.Type.valueOf(codec);
```

Comment:
this way of using Codec.Type makes code less flexible. It comes harder to write a unit/benchmark test, in this case.
Please consider passing Codec.Type in the constructor instead of using singletons

Reply:
This is coming from docopt, so a conversion is needed at some point. I don't want to overload the constructor in this patch, but in a subsequent patch I'd like to introduce StreamLogConfig class.

Reply:
LogUnitServerConfig already exists, in reality, the config is actually LUConfig + StreamLogConfig, so you can move the code into the config class

Reply:
Bet.

Reply:
again? :)